### PR TITLE
Updates Ceilometer template for Mitaka

### PR DESCRIPTION
**Pull request by sayalilunkad**

### CODE REVIEW

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 1]
```diff
@@ -1396,47 +1559,80 @@ telemetry_secret = <%= @metering_secret %>
 # From ceilometer
 #
 
-# User name to use for OpenStack service access. (string value)
-# Deprecated group/name - [DEFAULT]/os_username
-#os_username = ceilometer
-os_username = <%= @keystone_settings['service_user'] %>
```
Comment (aplanas) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: there is not anymore os_username / os_password?

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 2]
```diff
@@ -1396,47 +1559,80 @@ telemetry_secret = <%= @metering_secret %>
...
-#os_tenant_name = admin
-os_tenant_name = <%= @keystone_settings['service_tenant'] %>
+# Config Section from which to load plugin specific options (string value)
+#auth_section = <None>
 
-# Certificate chain for SSL validation. (string value)
-#os_cacert = <None>
+# Authentication URL (string value)
+#auth_url = <None>
+auth_url = <%= @internal_auth_url_v2 %>
```
Comment (aplanas) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: v3 doesn't work?

Comment (sayalilunkad) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: Not sure, but this just updated the template with the existing values. For this one I just used the value used for os_auth_url.

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb (hunk 1 above)
Comment (sayalilunkad) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: no now it is username and password as you see here https://github.com/crowbar/crowbar-openstack/pull/490/commits/3649cad2b2aca60cb0e1aeac1c917f569674537e#diff-8ba6ab0f73ac9ad5fa0f2fe85bb234deR1625

**Comment by toabctl** [comment]
Can you update the commit message please? Something like "Ceilometer: Update config template for Mitaka"

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 3]
```diff
@@ -989,22 +1087,30 @@ admin_tenant_name = <%= @keystone_settings['service_tenant'] %>
...
+
+# Number of seconds to wait before publishing sampleswhen batch_size is not
+# reached (None means indefinitely) (integer value)
+#batch_timeout = <None>
+
+# Number of workers for notification service, default value is 1. (integer
+# value)
 # Minimum value: 1
 # Deprecated group/name - [DEFAULT]/notification_workers
 #workers = 1
```
Comment (toabctl) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: I think we want to have ```<%= [node["cpu"]["total"], 2, 4].sort[1] %>``` here

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb (hunk 2 above)
Comment (toabctl) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: Can you check this? And change it in an extra commit to allo v3?

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 4]
```diff
@@ -1454,6 +1650,10 @@ insecure= <%= @keystone_settings['insecure'] %>
 # Neutron service type. (string value)
 #neutron = network
 
+# Neutron load balancer version. (string value)
+# Allowed values: v1, v2
+#neutron_lbaas_version = v2
```
Comment (toabctl) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: I think we currently only support lbaasv1 (see https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/neutron/templates/default/neutron_lbaas.conf.erb#L77 )

Comment (aplanas) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: Ah this is right. There is a patch from @vuntz to move to v2 (https://github.com/crowbar/crowbar-openstack/compare/stable/3.0...vuntz:f5-lbaas-3.0.patch)

Currently is not working on postgresql, but we are waiting this to be merged (https://review.openstack.org/#/c/350038/) (but the patch is in the package)

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb (hunk 3 above)
Comment (sayalilunkad) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: It is already there present in the line after.

Comment (sayalilunkad) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: It is already there present in the line after.

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 5]
```diff
@@ -1396,47 +1559,80 @@ telemetry_secret = <%= @metering_secret %>
...
+region_name = <%= @keystone_settings['endpoint_region'] %>
 
-# Password to use for OpenStack service access. (string value)
-# Deprecated group/name - [DEFAULT]/os_password
-#os_password = admin
-os_password = <%= @keystone_settings['service_password'] %>
+# Type of endpoint in Identity service catalog to use for communication with
+# OpenStack services. (string value)
+# Allowed values: public, internal, admin, auth, publicURL, internalURL, adminURL
+# Deprecated group/name - [service_credentials]/os_endpoint_type
```
Comment (rhafer) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: The default config in the package has ```# Deprecated group/name - [DEFAULT]/os_endpoint_type``` here, which one is correct? There are quite a few similar differences between this file and the one in the package.

**Comment by rhafer** [comment]
+1

**Comment by rhafer** [comment]
@aplanas Would you mind taking another look at this?

File: chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb [hunk 6]
```diff
@@ -740,16 +765,17 @@ connection = <%= @database_connection %>
...
 
-# Tolerance of IPMI/NM polling failures before disable this pollster.
-# Negative indicates retrying forever. (integer value)
+# Tolerance of IPMI/NM polling failures before disable this pollster. Negative
+# indicates retrying forever. (integer value)
 #polling_retry = 3
 
 
 [keystone_authtoken]
+signing_dir = /var/cache/ceilometer/keystone-signing
```
Comment (toabctl) [inline chef/cookbooks/ceilometer/templates/default/ceilometer.conf.erb]: Should be removed. It's already there a couple of lines later.

**Comment by aplanas** [comment]
I did a comparison with `grep -v "^#" ceilometer.conf.erb | cat -s`, but I did not test.

+1

**Comment by toabctl** [comment]
+1

**Pull request merged**